lib/core/src/chain/bitcoin.rs

```diff
@@ -60,9 +65,21 @@ impl BitcoinChainService for ElectrumClient {
             maybe_popped_header = Some(header)
         }
 
-        if let Some(popped_header) = maybe_popped_header {
-            let tip: HeaderNotification = popped_header.try_into()?;
-            self.tip = tip;
+        match maybe_popped_header {
+            Some(popped_header) => {
+                let tip: HeaderNotification = popped_header.try_into()?;
+                self.tip = tip;
+            }
+            None => {
```

Comment:
Does this mean it will resubscribe anytime we will call tip and there isn't any new block? If this is so then I am concern with unnecessary network overhead whenever we call tip.

Reply:
Understandable. I can dig into https://github.com/bitcoindevkit/rust-electrum-client to see if it's fixable there. It seems `block_headers_pop_raw()` does not check the stream state, so jit ust tries to pop the header from an array. Currently there needs to be some call to network see if the stream is closed and then re-establish the connection (then re-subscribe to the header notifications)